erigon-lib/kv/mdbx/kv_mdbx.go

```diff
@@ -696,16 +696,19 @@ func (db *MdbxKV) trackTxBegin() bool {
 }
 
 func (db *MdbxKV) hasTxsAllDoneAndClosed() bool {
+	db.txsCountMutex.Lock()
```

Comment:
I'd replace Mutex with RWMutex btw. Not related to the fix, but could increase performance.

Reply:
RW mutex isn't always a silver bullet, we changed some code to a normal mutex and saw a performance improvement.  I can't remember the details now, but we shouldn't use them everywhere.  Maybe we could benchmark them against each other in the future.  But definitely this needs doing in a case-by-case basis.